RallyHereIntegration/Source/RallyHereIntegration/Private/RH_SessionHelpers.h

```diff
@@ -852,3 +869,29 @@ class FRH_SessionJoinByPlatformIdHelper : public FRH_SessionLookupHelper
 
 	FRH_OnSessionUpdatedDelegateBlock Delegate;
 };
+
+
+// Wrapper around a delete that will remove the session locally once the delete is completed, without requiring a poll on the session
+class FRH_SessionLeaveHelper : public FRH_SessionRequestAndModifyHelper<RallyHereAPI::Traits_LeaveSessionByIdSelf>
+{
+public:
+	typedef RallyHereAPI::Traits_LeaveSessionByIdSelf BaseType;
+
+	FRH_SessionLeaveHelper(FRH_SessionOwnerPtr InSessionOwner, const FString& InSessionId, FRH_OnSessionUpdatedDelegateBlock InDelegate = FRH_OnSessionUpdatedDelegateBlock(), int32 InPriority = DefaultRallyHereAPIPriority)
+		: FRH_SessionRequestAndModifyHelper(InSessionOwner, InSessionId, InDelegate, InPriority)
+	{
+	}
+
+	virtual void OnSessionPollComplete(bool bSuccess, bool bResetTimer)
+	{
+		// ignore success flag on the poll (it can succeed or fail depending on the session state, we just want to verify that it removed the session from the owner)
+		RHSession = Cast<URH_JoinedSession>(SessionOwner->GetSessionById(SessionId));
+		Completed(!RHSession.IsValid());	// since this is a delete helper, we want the session to be INVALID at this point
```

Comment:
I expected you to do some kind of manual leave here.  As it is, how does this actually differ from reporting if the poll failed?  I guess said differently, doesn't this report failure if you send a leave, get invited, poll, then return the result of the poll? (that you are in the session?)

To make this correct, I'm nearly certain you have to explicitly allow polls to not "fail" if they get a 404.  As 404 is our way of determining that the session no longer exists.  On the other hand, if you expect a leave to fail if you end up in an invited state due to order of operations (the time window between the leave and the poll being filled with an invite), then I suppose this is okay?  How do you expect game teams to react to the return from this?

Reply:
The poll itself imports the new state (or lack thereof) and internally triggers the leave.  This way it works the same as if you were kicked from a session and polled and found out you were no longer a member.

Reply:
Sure, the question is what you would expect, as a game programmer, from this series of events

1. Make a request to leave session
2. Session API removes you
3. Different player re-invites you
4. Session API puts you back in the session
5. You finally poll after receiving the response from the leave from Session API
6. Session API returns that you are in the session

Your code currently returns that the leave session failed.

Reply:
Note that being in the invited state is effectively being a part of the session (leave call is also how you decline an invite)